File: 3379/main.cpp
class Solution
{
public:
    vector<int> constructTransformedArray(vector<int> &nums)
    {
        auto len = nums.size();
        auto res = vector<int>(len);

        for (int i = 0; i < len; i++)
        {
            res[i] = nums[i] ? (i + nums[i]) % len : 0;
        }

        return res;
    }
};

File: 3379/main.ts
function constructTransformedArray(nums: number[]): number[] {
	return nums.map((v, i) => v ? nums.at((i + v) % nums.length)! : 0);
}

/*   *   *   *   *   *   *   *   *   *   */
/*   *   *   *   *   *   *   *   *   *   */

import { assertEquals } from "@std/assert";

Deno.test("Case 1", () => {
	const nums = [3, -2, 1, 1];

	const $result = constructTransformedArray(nums);
	const $expect = [1, 1, 1, 3];

	assertEquals($result, $expect);
});

Deno.test("Case 2", () => {
	const nums = [-1, 4, -1];

	const $result = constructTransformedArray(nums);
	const $expect = [-1, -1, 4];

	assertEquals($result, $expect);
});
